### PR TITLE
feat: 0.3.x dev - Update dependencies and PHP to 8.1

### LINKED ISSUE #45: Add support for psr/log 2.0 or 3.0
### Problem

I want to add support for Laravel 10 within open food fact laravel package, and Laravel 10 now relies on psr/log:^2.0|^3.0, but since it relies on this package, and this package only supports psr/log:^1.0, it cannot be updated while this package is not updated as well.

### Proposed solution

Upgrade psr/log to ^2.0 or ^3.0 (or both).

### Part of

- https://github.com/openfoodfacts/openfoodfacts-laravel/issues/39

### PR DESCRIPTION
**Pull request by Dwarfex**
### What
- Update dependencies
- Update PHP to 8.1

### Fixes

#45

### CODE REVIEW

**Comment by khalyomede** [comment]
This pull request will fix https://github.com/openfoodfacts/openfoodfacts-php/issues/45.

**Comment by teolemon** [comment]
Can we merge?

**Review by epalmans — CHANGES_REQUESTED** [review]
all good to me, but I noticed tests failed due to a (seemingly unrelated?) problem with calls made out to  https://world.openfoodfacts.net/cgi/product_jqm2.pl : "Server error: `POST https://world.openfoodfacts.net/cgi/product_jqm2.pl` resulted in a `502 Bad Gateway` response".

Happens at calling https://github.com/openfoodfacts/openfoodfacts-php/blob/ef4eb31333286270b2d10e0e7393ecbf1ee15658/src/Api.php#L472 from:
- OpenFoodFactsTests\ApiFoodCacheTest::testApiAddProduct
- OpenFoodFactsTests\ApiFoodTest::testApiAddProduct

**Comment by scybulski** [comment]
What's the status here? I've just run all the tests and they pass so the error might have been just a temporary api issue not related to this repository. And as a side note it's blocking a related package https://github.com/openfoodfacts/openfoodfacts-laravel/pull/43 from being updated to the recent Laravel,

**Comment by DonatasIk** [comment]
This needs to be merged. It's creating a bottleneck for the related packages.

**Review by Benoit382 — APPROVED** [review]

File: src/Collection.php [hunk 1]
```diff
@@ -52,10 +48,10 @@ public function __construct(array $data = null, string $api = null)
             }
         }
 
-        $this->count    = $data['count'];
-        $this->page     = $data['page'];
-        $this->skip     = $data['skip'];
-        $this->pageSize = $data['page_size'];
+        $this->count    = $data['count'] ?? null;
```
Comment (Benoit382) [inline src/Collection.php]:
```suggestion
        $this->count    = $data['count'] ?? 0;
```
Error on this test

**Comment by Benoit382** [comment]
I launched the phpunit on my side, except this error : https://github.com/openfoodfacts/openfoodfacts-php/pull/47#discussion_r1583230170 all it's works

---
In this project, it's real difficult to update the code...

**Comment by teolemon** [comment]
@epalmans you need to reapprove since you requested changes

**Review by epalmans — APPROVED** [review]

**Comment by epalmans** [comment]
> @epalmans you need to reapprove since you requested changes

done.
Changed:
- suggested change from @Benoit382
- marked for skipping: the test that was failing due to api issues
- satisified cs-fixer/phpstan

After your review we may merge and tag it

**Review by Benoit382 — APPROVED** [review]

**Review by teolemon — APPROVED** [review]
Thank you all for getting this one through 🎉

**Pull request merged**